src/applications/personalization/dashboard-2/tests/unit/healthcare.unit.spec.jsx

```diff
@@ -1,22 +1,138 @@
+                data: {
+                  currentItem: {
+                    attributes: {
+                      unreadCount: 3,
+                    },
+                  },
+                },
+                ui: {
+                  nav: {
+                    foldersExpanded: false,
+                    visible: false,
+                  },
+                },
+              },
+            },
+          },
+        };
+      });
+      afterEach(() => {
+        resetFetch();
+      });
 
-    it('when enrolled in VA health care, should not render when they have never had any prescriptions managed by the VA', () => {
-      const wrapper = mount(<HealthCare {...defaultProps} />);
-      expect(wrapper.find('Prescriptions').exists()).to.be.false;
-      wrapper.unmount();
+      it('should attempt to get messaging data', async () => {
+        view = renderInReduxProvider(<HealthCare />, {
+          initialState,
+          reducers,
+        });
+        await wait(1);
+        const fetchCalls = global.fetch.getCalls();
+        // make sure we are fetching messaging folders
+        expect(
+          fetchCalls.some(call => {
+            return call.args[0].includes('v0/messaging/health/folders/0');
+          }),
+        ).to.be.true;
+      });
+
+      it('should render the unread messages count', async () => {
+        view = renderInReduxProvider(<HealthCare dataLoadingDisabled />, {
+          initialState,
+          reducers,
+        });
+        expect(
+          await view.findByText(new RegExp(`you have 3 new messages`, 'i')),
+        ).to.exist;
+      });
+    });
+
+    context('when user lacks the `messaging` service', () => {
+      beforeEach(() => {
+        mockFetch();
+        initialState = {
+          user: {
+            profile: {
+              services: [],
+            },
+          },
+          health: {
+            appointments: {
+              data: [],
+            },
+            msg: {
+              folders: {
+                data: {
+                  currentItem: {
+                    attributes: {
+                      unreadCount: 0,
+                    },
+                  },
+                },
+                ui: {
+                  nav: {
+                    foldersExpanded: false,
+                    visible: false,
+                  },
+                },
+              },
+            },
+          },
+        };
+      });
+      afterEach(() => {
+        resetFetch();
+      });
+      it('should not attempt to get messaging data', async () => {
+        view = renderInReduxProvider(<HealthCare />, {
+          initialState,
+          reducers,
+        });
+        await wait(1);
+        const fetchCalls = global.fetch.getCalls();
+        // make sure we are not fetching messaging folders
+        expect(
+          fetchCalls.some(call => {
+            return call.args[0].includes('v0/messaging/health/folders/0');
+          }),
+        ).to.be.false;
```

Comment:
we should really wrap all this up in an easy-to-use helper. Maybe `shouldNotCallApi('v0/messaging/health/folders/0')`

Reply:
ticked added https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/21730